### PR TITLE
Feature/add add product section and add product page

### PR DESCRIPTION
**Pull request by EmanSayed94**

### CODE REVIEW

File: src/components/AddProductSection/index.jsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  countItemsChange,
+  tabs,
+  currentTabe,
+  onAddButtonClicked,
+  onCancelButtonClicked,
+}) => {
+  return (
+    <React.Fragment>
+      <Header />
+
```
Comment (ahmedazizsq) [inline src/components/AddProductSection/index.jsx:29]: remove extra vertical space

File: src/components/AddProductSection/index.jsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  onChange,
+  onPaymentTypeChange,
+  onCategoryChange,
+  countItemsChange,
+  tabs,
+  currentTabe,
+  onAddButtonClicked,
+  onCancelButtonClicked,
+}) => {
+  return (
```
Comment (ahmedazizsq) [inline src/components/AddProductSection/index.jsx:26]: remove return as there is no logic before rendering

File: src/components/Input/index.jsx [hunk 3]
```diff
@@ -6,9 +6,10 @@ const Input = ({
   className,
   type,
   value,
+  checked,
   onChange,
   placeholder,
-  id
+  id,
 }) => {
   return (
```
Comment (ahmedazizsq) [inline src/components/Input/index.jsx:14]: remove return

File: src/components/Textarea/index.jsx [hunk 4]
```diff
@@ -7,12 +7,13 @@ import React from "react";
         ></Textarea> */
```
Comment (ahmedazizsq) [inline src/components/Textarea/index.jsx]: remove comment

File: src/containers/AddProductPage/index.jsx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+      { id: 5, name: "PayPal", checked: false },
+    ],
+  };
+
+  handleChange = (event) => {
+    const product = { ...this.state.product };
+    product[event.target.name] = event.target.value;
+    console.log(product);
+    this.setState({ product });
+  };
```
Comment (ahmedazizsq) [inline src/containers/AddProductPage/index.jsx:31]: add new line

File: src/containers/AddProductPage/index.jsx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+  handleCategoryChange = (e) => {
+    const product = { ...this.state.product };
+    const categories = [...this.state.categories];
+    const categorySelected = parseInt(e.target.value);
+    product.category = categories.find(
+      (category) => category.id === categorySelected
+    );
+    this.setState({ product });
+  };
```
Comment (ahmedazizsq) [inline src/containers/AddProductPage/index.jsx:51]: add new line

File: src/containers/App/index.jsx [hunk 7]
```diff
@@ -1,9 +1,17 @@
 import React, { Component } from "react";
+import AddProducSection from "./../../components/AddProductSection/index";
+
```
Comment (ahmedazizsq) [inline src/containers/App/index.jsx]: remove extra line

File: src/containers/App/index.jsx [hunk 8]
```diff
@@ -1,9 +1,17 @@
...
+import AddProduct from "./../AddProductPage/index";
+// import HistoryFromUser HistoryFromUser
+import HistoryFromUser from "./../HistoryFromUserPage/index";
 
 class App extends Component {
   render() {
     return (
       <React.Fragment>
+        {/* <AddProducSection /> */}
```
Comment (ahmedazizsq) [inline src/containers/App/index.jsx]: remove comment

Comment (EmanSayed94) [reply src/containers/App/index.jsx]: my app is empty !! how import and comment exist

File: src/components/AddProductSection/Partials/productDetails.jsx [hunk 9]
```diff
@@ -16,96 +16,94 @@ const ProductDetails = ({
...
-        {paymentTypes.map((paymentType) => (
-          <div className="paymentItem">
-            <Input
-              key={paymentType.id}
-              type="checkbox"
-              className="paymentItem__input"
-              onChange={onPaymentTypeChange}
-              name={paymentType.name}
-              checked={paymentType.checked}
-            />
```
Comment (halaelsaka2) [inline src/components/AddProductSection/Partials/productDetails.jsx]: each input should have value

File: src/components/AddProductSection/Partials/productDetails.jsx [hunk 10]
```diff
@@ -16,96 +16,94 @@ const ProductDetails = ({
...
+      {paymentTypes.map((paymentType) => (
+        <div className="paymentItem">
+          <Input
+            key={paymentType.id}
+            type="checkbox"
+            className="paymentItem__input"
+            onChange={onPaymentTypeChange}
+            name={paymentType.name}
+            checked={paymentType.checked}
+          />
```
Comment (halaelsaka2) [inline src/components/AddProductSection/Partials/productDetails.jsx]: each input should have value

File: src/components/AddProductSection/Partials/productDetails.jsx [hunk 11]
```diff
@@ -16,96 +16,94 @@ const ProductDetails = ({
...
+}) => (
+  <div className="productDetailsContainer">
+    <label>Product Name</label>
+    <Input
+      className="input input--m"
+      placeholder="Product name"
+      name="name"
+      onChange={onChange}
+      value={productNameValue}
+    />
```
Comment (halaelsaka2) [inline src/components/AddProductSection/Partials/productDetails.jsx:28]: each input should have type

**Pull request merged**